projects/plugins/boost/app/class-jetpack-boost.php

```diff
@@ -520,6 +520,7 @@ public function get_admin_notices() {
 	 * still capture the change of environment event and flag Critical CSS for a rebuild.
 	 */
 	public function handle_theme_change() {
+		do_action( 'jetpack_boost_clear_critical_css' );
```

Comment:
We should do the action `jetpack_boost_clear_cache` instead, considering the following comment.

Reply:
Can you explain it in brief?

Reply:
So your implementation works like this:
1. Theme changed
2. `do_action( 'handle_theme_change' );`
3. `handle_theme_change()` gets called from action: `handle_theme_change`.
4. That fires `do_action( 'jetpack_boost_clear_critical_css' );`
5. `clear_critical_css()` gets called from action: `jetpack_boost_clear_critical_css`.

The suggested solution follows this flow:
1. Theme changed.
2. `do_action( 'handle_theme_change' );`
5. `clear_critical_css()` gets called from action `handle_theme_change`.


The reasoning behind this is actions should represent the events that happened or happening, the action should not care about what tasks need to be done.

Keeping this in mind, the hook names should be improved for the plugin. However, that is a task of its own and outside the scope of this PR. The suggested solution is just a step in the right direction.

Reply:
> public function handle_theme_change() {
>         $critical_css = new Critical_CSS();
>         $critical_css->clear_critical_css();
> 		Admin::clear_dismissed_notice( Regenerate_Admin_Notice::SLUG );
> 		\update_option( Critical_CSS::RESET_REASON_STORAGE_KEY, Regenerate_Admin_Notice::REASON_THEME_CHANGE, false );
> }

@eAdnan007, this is how changes are needed to be done?

Reply:
Hi @eAdnan007, I got it. 💯 

> The suggested solution follows this flow:
> 
> Theme changed.
> do_action( 'handle_theme_change' );
> clear_critical_css() gets called from action handle_theme_change

It was great learning for me on `add_action` `do_action` hooks! 🥳